### ISSUE TITLE
disk_encryption table misreporting on macOS

### ISSUE THREAD

**Issue by FritzX6** [osquery/osquery#6718]
### Description of issue:
Recently a user in the osquery Slack reported inconsistent behavior with the `disk_encryption` table reporting drives as encrypted when FileVault2 was not enabled.

### Link to Osquery Conversation:
https://osquery.slack.com/archives/C08VA8R6F/p1602795821048800

### Quoted Relevant Text:

> **Julian Scala**
> I had a similar query, but got it returning encrypted=1 on devices with FileVault service deactivated
> 
> **Fritz**
> That is very curious.
> So they were unencrypted volumes reporting back as encrypted?
> Or they were encrypted using something other than FileVault?
> 
> **Julian Scala**
> ```
> select
> case when de.encrypted = 1 then "true" else "false" end as filevault,
> from mounts m
> join disk_encryption de on m.device_alias = de.name
> where m.path = '/';
> ```
> Devices unencrypted reporting back as encrypted.

### Example Screenshot of Issue:
![image](https://user-images.githubusercontent.com/931724/96278650-bd015880-0fa3-11eb-879f-bd3a97469d1c.png)

### OS versions behavior was observed on:
```
10.14.5
10.15.6
10.15.7
```

### Osquery versions behavior was observed on:
```
3.3.2
4.4.0
```

### Steps to Reproduce
Not sure... will edit issue if I can determine reliable steps to reproduce

### Speculation to Cause
I am wondering if all APFS devices are simply being interpreted as `Encrypted=True` even if FileVault is not configured/enabled.

**Comment by jlopezscala** [comment]
This is being returned on one of the devices on

`diskutil apfs list`

```
[X]Guest10/19 12:38:46
APFS Container (1 found)
|
+-- Container disk1 48574CD8-F9A0-45D0-89AA-CBCE79CB930A
    ====================================================
    APFS Container Reference:     disk1
    Size (Capacity Ceiling):      499963174912 B (500.0 GB)
    Capacity In Use By Volumes:   387407650816 B (387.4 GB) (77.5% used)
    Capacity Not Allocated:       112555524096 B (112.6 GB) (22.5% free)
    |
    +-< Physical Store disk0s2 04920AA2-0CFE-4C96-8452-FA38ABB9347F
    |   -----------------------------------------------------------
    |   APFS Physical Store Disk:   disk0s2
    |   Size:                       499963174912 B (500.0 GB)
    |
    +-> Volume disk1s1 27C143C8-E6BB-4DD9-9B1B-733AC64EBA40
    |   ---------------------------------------------------
    |   APFS Volume Disk (Role):   disk1s1 (No specific role)
    |   Name:                      Macintosh HD (Case-insensitive)
    |   Mount Point:               /
    |   Capacity Consumed:         375944740864 B (375.9 GB)
    |   FileVault:                 No (Encrypted at rest)
    |
    +-> Volume disk1s2 D7749C98-1757-479E-A112-09333D1203D5
    |   ---------------------------------------------------
    |   APFS Volume Disk (Role):   disk1s2 (Preboot)
    |   Name:                      Preboot (Case-insensitive)
    |   Mount Point:               Not Mounted
    |   Capacity Consumed:         45330432 B (45.3 MB)
    |   FileVault:                 No
    |
    +-> Volume disk1s3 AEFAAA48-2A4D-49DF-92D5-D2D605655AB1
    |   ---------------------------------------------------
    |   APFS Volume Disk (Role):   disk1s3 (Recovery)
    |   Name:                      Recovery (Case-insensitive)
    |   Mount Point:               /Volumes/Recovery
    |   Capacity Consumed:         518598656 B (518.6 MB)
    |   FileVault:                 No
    |
    +-> Volume disk1s4 E09331A2-9AFE-43E1-86FA-0E59CE660D2E
        ---------------------------------------------------
        APFS Volume Disk (Role):   disk1s4 (VM)
        Name:                      VM (Case-insensitive)
        Mount Point:               /private/var/vm
        Capacity Consumed:         10737512448 B (10.7 GB)
        FileVault:                 No (Encrypted at rest)
```

**Comment by Smjert** [comment]
This seems to be an old issue: https://github.com/osquery/osquery/issues/5223

**Comment by jlopezscala** [comment]
Pretty sure this can be closed after this: #6823 🚀
